app/routes/events/components/StripeElement.js

```diff
@@ -199,112 +245,82 @@ class PaymentRequestForm extends Component<
-      this.completePayment(clientSecret);
-    }
-    if (paymentError) {
-      this.completePaymentManual('fail');
-      this.props.setError(paymentError);
-    }
-  }
-
-  componentWillUnmount() {
-    /*
-     * If the component unmounts, the registration will have updated,
-     * and the user will not be able to pay.
-     * This can be because the payment updated in the backend,
-     * or the user has unregistered. In the rare case that the payment has started
-     * processing, we cancel just so the user does not have to wait until the
-     * payment request times out.
-     *
-     */
-    this.completePaymentManual('fail');
-  }
-
-  completePayment = async (clientSecret) => {
-    const { stripe } = this.props;
-    const { paymentMethodId, complete } = this.state;
-
-    if (!complete || !paymentMethodId) {
-      return;
+  }, [
+    paymentRequest,
+    clientSecret,
+    stripe,
+    event,
+    completePayment,
+    createPaymentIntent,
+    setCanPaymentRequest,
+  ]);
+
+  useEffect(() => {
+    if (clientSecret && completePayment && !paymentStarted) {
+      setPaymentStarted(true);
+      completePayment(clientSecret);
     }
+  }, [clientSecret, completePayment, paymentStarted, completePaymentManual]);
 
-    const { error: confirmError } = await stripe.confirmPaymentIntent(
-      clientSecret,
-      {
-        payment_method: paymentMethodId,
-      }
-    );
-    if (confirmError) {
-      complete('fail');
-      return;
-    }
-    complete('success');
-    this.props.setLoading(true);
-
-    const { error } = await stripe.handleCardPayment(clientSecret);
-    if (error) {
-      this.props.setError(error.message);
-    } else {
-      this.props.setSuccess();
-    }
-    this.props.setLoading(false);
-  };
-
-  completePaymentManual = async (status: CompleteStatus) => {
-    const { complete } = this.state;
-
-    if (!complete) {
-      return;
-    }
-
-    complete(status);
+  useEffect(() => {
+    return () => {
+      completePaymentManual('fail');
+    };
+  }, []);
 
-    if (status === 'success') {
-      this.props.setSuccess();
+  useEffect(() => {
+    if (paymentError && setError && completePaymentManual) {
+      completePaymentManual('fail');
+      setError(paymentError);
     }
-  };
-
-  render() {
-    return (
-      <div style={{ flex: 1 }}>
-        {this.state.canMakePayment && (
-          <PaymentRequestButtonElement
-            paymentRequest={this.state.paymentRequest}
-            className={stripeStyles.PaymentRequestButton}
-            style={{
+  }, [paymentError, completePaymentManual, setError]);
+
+  return (
+    <div style={{ flex: 1 }}>
+      {canMakePayment && paymentRequest && (
+        <PaymentRequestButtonElement
+          onClick={(e) => {
+            if (paymentMethod) {
+              e.preventDefault();
+              setError(
+                'Error while processing payment method. Please refresh the page and try again.'
```

Comment:
I think this error message should be written in Norwegian, as it would make it more "at home" in an otherwise Norwegian website 🇳🇴